File: typescript-logistic-benchmark/src/actions/RepeatAction.ts
import { calculate } from '../core';
import { IParameters } from '../parameters/IParameters';
import { Action } from './IAction';

export class RepeatAction implements Action {
  run(params: IParameters): void {
    if (params.repetitions) {
      const times = new Array<number>(params.repetitions);

      const t0 = Date.now();
      for (let i = 0; i < params.repetitions; i++) {
        process.stdout.write(`\r${i + 1}\t/\t${params.repetitions}`);
        times[i] = calculate(params.x0, params.r, params.interactions).time;
      }
      const time = Date.now() - t0;
      process.stdout.write("\n");

      const average = times.reduce((a, b) => a + b, 0) / params.repetitions;

      console.log("AVERAGE", average, "ms");
      console.log("TOTAL_TIME " + time)
    } else {
      throw new Error("Repetitions not defined!");
    }
  }
}

File: java-logistic-benchmark/logistic-benchmark/src/main/java/rafael/logistic_benchmark/actions/RepeatAction.java
package rafael.logistic_benchmark.actions;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import rafael.logistic_benchmark.core.Processor;
import rafael.logistic_benchmark.parameters.Parameters;

import java.util.OptionalDouble;
import java.util.stream.IntStream;
import java.util.stream.LongStream;

public class RepeatAction implements Action {

    private static final Logger LOGGER = LoggerFactory.getLogger(SimpleAction.class.getName());

    Processor processor = new Processor();

    @Override
    public void run(Parameters parameters) {
        long[] times = new long[parameters.getRepititions()];

        // warming
        processor.calculate(parameters.getX0(), parameters.getR(), parameters.getInteractions());

        long t0 = System.currentTimeMillis();
        IntStream
                .rangeClosed(1, parameters.getRepititions())
                .peek(i -> System.out.printf("\r%5d / %5d", i, parameters.getRepititions()))
                .forEach(i -> times[i - 1] = processor.calculate(parameters.getX0(), parameters.getR(), parameters.getInteractions()).time());
        long time = System.currentTimeMillis() - t0;
        System.out.println();

//        LongStream.of(times).forEach(System.out::println);
        OptionalDouble average = LongStream.of(times).average();
        LOGGER.info("AVERAGE {} ms", average.getAsDouble());
        System.out.println("TOTAL_TIME " + time);
    }
}